src/django_components/component.py

```diff
@@ -253,11 +255,19 @@ def get_template(self, context: Mapping) -> Template:
             f"Note: this attribute is not required if you are overriding the class's `get_template*()` methods."
         )
 
-    def render_from_input(self, context: Context, args: Union[List, Tuple], kwargs: Dict) -> str:
-        component_context: dict = self.get_context_data(*args, **kwargs)
+    def render_from_input(self, context: Context, args: Union[List, Tuple], kwargs: Dict[str, Any]) -> str:
+        processed_kwargs, attrs = extract_attrs_from_component_input(kwargs)
+        # TODO: Change `get_context_data` signature so we can distinguish
+        #   input named "attrs" from the fallthrough attrs:
+        #   `get_context_data(kwargs: Dict[Any, Any], attrs: Dict[str, Any])`
+        component_context: dict = self.get_context_data(*args, attrs=attrs, **processed_kwargs)
```

Comment:
Passing `attrs` to `get_context_data` is not idea, because practically it's not possible to distinguish between the attributes and a component input named `attrs`.

IMO a better approach would be to pass the component inputs as a dict (instead of spreading it with `**kwargs`). For this, we'd have to remove the option to pass positional arguments (`*args`). So this would be a breaking change. But IMO this would be the right way forward, because there's more features for which we'd need to make this change, like the "Typed component inputs" in https://github.com/EmilStenstrom/django-components/issues/483.

So instead of
```py
class MyClass(component.Component):
	def get_context_data(self, *args: Any, **kwargs: Any):
		return {...}
```

we'd have

```py
class MyClass(component.Component):
	def get_context_data(self, props: Dict[str, Any], attrs: Dict[str, Any]):
		return {...}
```

Furthermore, we could go a step further and future-proof the function signature in a similar manner how Vue does it, where the last argument would be an object that would serve as a container for any possible future fields.

```py
# in django_components/component.py
class ComponentContext(NamedTuple):
	attrs: Dict[str, Any]

# in my_component.py
class MyComponent(component.Component):
	def get_context_data(self, props, ctx)
		# Access attrs like this:
		ctx.attrs
```